src/aiod/authorisation/authorisation.py

```diff
@@ -0,0 +1,80 @@
+import os
+from keycloak import KeycloakOpenID
+
+from aiod.config.settings import server_url, client_id, realm
+
+keycloak_openid = KeycloakOpenID(
+    server_url=server_url,
+    client_id=client_id,
+    realm_name=realm,
+)
+
+
+def login(username: str, password: str) -> None:
+    """
+    Logs in the user with the provided username and password.
+
+    Args:
+        username (str): The username of the user.
+        password (str): The password of the user.
+
+    Raises:
+        MissingCredentialsError: If the username or password is missing.
+        MissingTokenError: If Keycloak returns an invalid token.
+    """
+    if username is None or password is None:
+        raise MissingCredentialsError(
+            "Username and/or password missing! Please provide your credentials and try again."
+        )
+    token = keycloak_openid.token(username, password)
+
+    if not token:
+        raise MissingTokenError("Keycloak returned an invalid token!")
```

Comment:
Is this default behavior if incorrect credentials are passed? Or when you fail to connect to the auth server? Other reasons? This error message is very technical and the regular user will probably not understand what they can do about this.

E.g., `FailedAuthenticationError` would already be better. If we can specify _why_ it (likely) failed in the error message, it would be even better.

Reply:
Good point. I caught the error raised by keycloak and included your suggestion.